vector/src/main/java/im/vector/util/DinsicUtils.java

```diff
@@ -210,4 +240,135 @@ public void onClick(DialogInterface dialog, int id) {
 
     }
 
+    //=============================================================================================
+    // Handle existing direct chat room
+    //=============================================================================================
+
+    /**
+     *
+     * Select the most suitable direct chat for the provided user identifier (matrix or third party id).
+     * During this search, the pending invite are considered too.
+     * The selected room (if any) may be opened synchronously or not. It depends if the room is ready or not.
+     *
+     * @param activity current activity
+     * @param participantId : participant id (matrix id ou email)
+     * @param session current session
+     * @param canCreate create the direct chat if it does not exist.
+     * @return boolean that says if the direct chat room is opened or not
+     *
+     */
+    public static boolean openDirectChat(final RiotAppCompatActivity activity, String participantId, final MXSession session, boolean canCreate) {
+        Room existingRoom = VectorRoomCreationActivity.isDirectChatRoomAlreadyExist(participantId, session, true);
+        boolean succeeded = false;
+
+        // direct message api callback
+        ApiCallback<String> createDirectMessageCallBack = new ApiCallback<String>() {
```

Comment:
rename this callback prepareDirectMessageCallBack, and use it to join the room too